### ISSUE TITLE
REST API response with /v1/unlockwallet

### ISSUE THREAD

**Issue by normoes** [lightningnetwork/lnd#3983]
### Background

I try to automate the unlocking of the wallet. I do this using the REST API and the `/v1/unkockwallet` endpoint.

As a result, I expected a HTTP response with a `200 (OK)` status code.

I received `503` instead with the body:
```
{"error":"transport is closing","message":"transport is closing","code":14}
```

**However, the wallet still was successfully unlocked.**

### Your environment

* `v0.9.0-beta`
* I run lnd within a docker container based on `golang:1.12.13-alpine3.10`.
  - `Linux 17649df81af1 4.15.0-74-generic #84-Ubuntu SMP Thu Dec 19 08:06:28 UTC 2019 x86_64 Linux`
* `bitcoind` version is ` v0.19.0.1`.

### Steps to reproduce

I start lnd and wait until it wants me to unlock the wallet.

I do:
```
curl -i -X POST --cacert /data/lnd/tls.cert   https://some_domain:8080/v1/unlockwallet -d '{"wallet_password": base64_encoded_passwd}'
```

### Expected behaviour

A HTTP response with a status code `200`.
And maybe a body in JSON format that says `{"result": "success"}`.

### Actual behaviour

I receive:
```
HTTP/1.1 503 Service Unavailable
Content-Type: application/json
Date: Tue, 04 Feb 2020 16:27:25 GMT
Content-Length: 75

{"error":"transport is closing","message":"transport is closing","code":14}
```

**And the wallet gets unlocked.**

I even tried including the default `readonly.macaroon` which gave the same status code `503` but did not return a body:
```
HTTP/1.1 503 Service Unavailable
Content-Type: application/json
Date: Wed, 05 Feb 2020 13:52:51 GMT
Content-Length: 75
```

Also, when providing a wrong password, the REST server returns:
```
HTTP/1.1 500 Internal Server Error
Content-Type: application/json
Trailer: Grpc-Trailer-Content-Type
Date: Wed, 05 Feb 2020 13:51:20 GMT
Transfer-Encoding: chunked

{"error":"invalid passphrase for master public key","message":"invalid passphrase for master public key","code":2}
```

which is fine in general. I would prefer another status code, though. Something like `400` maybe (just my personal preference).

**Comment by guggero** [comment]
Does this happen always or just sometimes? And what does the log of `lnd` say when this occurs? Could be a timing issue, because the `WalletUnlocker` is shut down just after the wallet is unlocked. I think this is related to #3982 and can probably be fixed at the same time.

**Comment by dethos** [comment]
I was able to reproduce the issue locally. It happens every time. 

Here is my `lnd` log, before and after getting the `{"error":"transport is closing","message":"transport is closing","code":14}` response from the REST API:
```
2020-02-05 17:26:26.848 [INF] LTND: Version: 0.9.0-beta commit=v0.9.0-beta, build=production, logging=default
2020-02-05 17:26:26.848 [INF] LTND: Active chain: Bitcoin (network=testnet)
2020-02-05 17:26:27.492 [INF] CHDB: Checking for schema update: latest_version=12, db_version=12
2020-02-05 17:26:27.504 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2020-02-05 17:26:27.505 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2020-02-05 17:26:27.505 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2020-02-05 17:26:38.614 [INF] LNWL: Opened wallet
2020-02-05 17:26:38.615 [INF] LTND: Primary chain is set to: bitcoin
2020-02-05 17:26:38.636 [INF] LTND: Initializing bitcoind backed fee estimator
2020-02-05 17:26:38.636 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on 127.0.0.1:28332
2020-02-05 17:26:38.636 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on 127.0.0.1:28333
```

In case it is helpful, I tried the same with an older version of `lnd` (0.8.1) the behavior is the same but the response I get is different: 
```
curl -i -X POST --cacert ~/.lnd/tls.cert https://localhost:8080/v1/unlockwallet -d '{"wallet_password": "<wallet-password-base64>"}'
HTTP/1.1 408 Request Timeout
Content-Type: application/json
Date: Wed, 05 Feb 2020 16:59:30 GMT
Content-Length: 37

{"error":"context canceled","code":1}
```

**Comment by Roasbeef** [comment]
> Could be a timing issue, because the WalletUnlocker is shut down just after the wallet is unlocked. I think this is related to #3982 and can probably be fixed at the same tim

Yeh I think this is the issue. Perhaps we can try to block until the message is received by the other end _then_ shutdown the unlocker service? Long-term I think we aim to keep the service running at all times so users can easily tell if the wallet is unlocked or not.

**Comment by normoes** [comment]
Thanks @dethos and @guggero 

Let's wait for the https://github.com/lightningnetwork/lnd/issues/3982 and test again.

**Comment by guggero** [comment]
This should be fixed by #3982. Feel free to reopen if this still happens.
